extensions/amp-story/1.0/amp-story.js

```diff
@@ -410,8 +410,17 @@ export class AmpStory extends AMP.BaseElement {
     // prerendering, because of a height incorrectly set to 0.
     this.mutateElement(() => {});
 
-    const pageEl = this.element.querySelector('amp-story-page');
-    pageEl && pageEl.setAttribute('active', '');
+    let page = this.element.querySelector('amp-story-page');
+    const maybePageId = parseQueryString(this.win.location.hash)['page'];
+    const maybePage =
+      maybePageId &&
+      this.element.querySelector(
+        `amp-story-page#${escapeCssSelectorIdent(maybePageId)}`
+      );
+    if (maybePage) {
+      page = maybePage;
+    }
```

Comment:
```suggestion
const page = this.element.querySelector(`amp-story-page#${escapeCssSelectorIdent(this.getInitialPageId_())}`)
```

Reply:
If the pageId passed does not exist, then we need to fallback on the first page as the active page. For that reason your snippet wouldn't work 100%, but I'll simplify it in a similar manner to make it fallback properly